builtin/providers/azurerm/resource_arm_managed_image_test.go

```diff
@@ -0,0 +1,345 @@
+		if err != nil {
+			return fmt.Errorf("Bad: Get on vmClient: %s", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound && shouldExist {
+			return fmt.Errorf("Bad: VM %q (resource group %q) does not exist", vmName, resourceGroup)
+		}
+		if resp.StatusCode != http.StatusNotFound && !shouldExist {
+			return fmt.Errorf("Bad: VM %q (resource group %q) still exists", vmName, resourceGroup)
+		}
+
+		log.Printf("[INFO] testing MANAGED IMAGE VM EXISTS - END.")
+
+		return nil
+	}
+}
+
+func testCheckAzureRMManagedImageDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).diskClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_managed_disk" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Managed Disk still exists: \n%#v", resp.Properties)
+		}
+	}
+
+	return nil
+}
+
+var testAccAzureRMManagedImage_standaloneImage = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%d"
+    location = "West Central US"
+}
+
+resource "azurerm_managed_image" "test" {
+    name = "accteste-%d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+	os_disk {
+		os_disk_ostype = "Linux"
+		os_disk_osstate = "Generalized"
+		os_disk_blob_uri = "https://terraformdev.blob.core.windows.net/packerimages/ubuntu_plain.vhd"
+ 	    os_disk_size_gb = 30
+	}
+
+    tags {
+        environment = "acctest"
+        cost-center = "ops"
+    }
+}`
+
+var testAccAzureRMManagedImage_customImage_fromVHD = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%d"
+    location = "West Central US"
+}
+
+resource "azurerm_virtual_network" "test" {
+    name = "acctvn-%d"
+    address_space = ["10.0.0.0/16"]
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_subnet" "test" {
+    name = "acctsub-%d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    virtual_network_name = "${azurerm_virtual_network.test.name}"
+    address_prefix = "10.0.2.0/24"
+}
+
+resource "azurerm_network_interface" "test" {
+    name = "acctni-%d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+
+    ip_configuration {
+    	name = "testconfiguration1"
+    	subnet_id = "${azurerm_subnet.test.id}"
+    	private_ip_address_allocation = "dynamic"
+    }
+}
+
+resource "azurerm_managed_image" "test" {
+    name = "accteste"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+	os_disk {
+		os_disk_ostype = "Linux"
+		os_disk_osstate = "Generalized"
+		os_disk_blob_uri = "https://terraformdev.blob.core.windows.net/packerimages/ubuntu_plain.vhd"
```

Comment:
This VHD is in my subscription - I can make it available for testing. Generally, right now, the testers would need to have resources to create images in their subscription (see comment about {subscription_id} above, not sure if we can do better on automating these tests.